doc/api/stream.md

```diff
@@ -2,22 +2,44 @@
 
     Stability: 2 - Stable
 
-A stream is an abstract interface implemented by various objects in
-Node.js. For example a [request to an HTTP server][http-incoming-message] is a
-stream, as is [`process.stdout`][]. Streams are readable, writable, or both. All
-streams are instances of [`EventEmitter`][].
-
-You can load the Stream base classes by doing `require('stream')`.
-There are base classes provided for [Readable][] streams, [Writable][]
-streams, [Duplex][] streams, and [Transform][] streams.
-
-This document is split up into 3 sections:
-
-1. The first section explains the parts of the API that you need to be
-   aware of to use streams in your programs.
-2. The second section explains the parts of the API that you need to
-   use if you implement your own custom streams yourself. The API is designed to
-   make this easy for you to do.
+The `stream` module provides the abstract API for working with streaming data
+within Node.js. The streams API is implemented by multiple objects within
+Node.js. For instance, a [request to an HTTP server][http-incoming-message] and
+[`process.stdout`][] are each stream instances.
+
+Streams can be readable, writable, or both. All streams are instances of
+[`EventEmitter`][].
+
+The `stream` module can be accessed using:
+
+```js
+const stream = require('stream');
+```
+
+The `stream` module is most useful for developer's creating new types of stream
+instances. Developer's consuming stream objects will rarely (if ever) have need
+to access the `stream` module directly.
+
+## Types of Streams
+
+There are four fundamental stream types within Node.js:
+
+* [Readable][] - streams from which data can be read (for example
+  [`process.stdin`][])
+* [Writable][] - streams to which data can be written (for example
+  [`process.stdout`][])
+* [Duplex][] - streams that are both Readable and Writable
```

Comment:
You can say that a socket is a duplex. A Duplex is (usually) a a form of bidirectional communication with another process.